includes/Generator/OrderAttribution.php

```diff
@@ -0,0 +1,321 @@
+				'_wc_order_attribution_source_type' => $source_type,
+			);
+		} else {
+			$meta = array(
+				'_wc_order_attribution_origin'             => $origin,
+				'_wc_order_attribution_device_type'        => $device_type,
+				'_wc_order_attribution_user_agent'         => self::get_random_user_agent_for_device( $device_type ),
+				'_wc_order_attribution_session_count'      => wp_rand( 1, 10 ),
+				'_wc_order_attribution_session_pages'      => wp_rand( 1, 10 ),
+				'_wc_order_attribution_session_start_time' => self::get_random_session_start_time( $order ),
+				'_wc_order_attribution_session_entry'      => $product_url,
+				'_wc_order_attribution_utm_content'        => $utm_content,
+				'_wc_order_attribution_utm_source'         => self::get_source( $source_type ),
+				'_wc_order_attribution_referrer'           => self::get_referrer( $source_type ),
+				'_wc_order_attribution_source_type'        => $source_type,
+			);
+		}
+
+		// If the source type is not typein ( Direct ), set a random utm medium.
+		if ( 'typein' !== $source_type ) {
+			$meta['_wc_order_attribution_utm_medium'] = self::get_random_utm_medium();
+		}
+
+		foreach ( $meta as $key => $value ) {
+			$order->add_meta_data( $key, $value );
+		}
+	}
+
+	/**
+	 * Get a random referrer based on the source type.
+	 *
+	 * @param string $source_type The source type.
+	 * @return string The referrer.
+	 */
+	public static function get_referrer( string $source_type ) {
+		// Set up the label based on the source type.
+		switch ( $source_type ) {
+			case 'utm':
+				$utm = array(
+					'https://woo.com/',
+					'https://twitter.com',
+				);
+				return $utm[ array_rand( $utm ) ];
+			case 'organic':
+				$organic = array(
+					'https://google.com',
+					'https://bing.com',
+				);
+				return $organic[ array_rand( $organic ) ];
+			case 'referral':
+				$refferal = array(
+					'https://woo.com/',
+					'https://facebook.com',
+					'https://twitter.com',
+				);
+				return $refferal[ array_rand( $refferal ) ];
+			case 'typein':
+				return '';
+			case 'admin':
+				return '';
+			default:
+				return '';
+		}
+	}
+
+	/**
+	 * Get a random utm medium.
+	 *
+	 * @return string The utm medium.
+	 */
+	public static function get_random_utm_medium() {
+		$utm_mediums = array(
+			'referral',
+			'cpc',
+			'email',
+			'social',
+			'organic',
+			'unknown',
+		);
+
+		return $utm_mediums[ array_rand( $utm_mediums ) ];
+	}
+
+	/**
+	 * Get the origin.
+	 *
+	 * @param string $source_type The source type.
+	 * @param string $source The source.
+	 *
+	 * @return string The origin.
+	 */
+	public static function get_origin( string $source_type, string $source ) {
+		// Set up the label based on the source type.
+		switch ( $source_type ) {
+			case 'utm':
+				return 'Source: ' . $source;
+			case 'organic':
+				return 'Organic: ' . $source;
+			case 'referral':
+				return 'Referral: ' . $source;
+			case 'typein':
+				return 'Direct';
+			case 'admin':
+				return 'Web admin';
```

Comment:
To be ready for https://github.com/woocommerce/woocommerce/pull/44519
```suggestion
			case 'admin':
				return 'Web admin';
			case 'mobile_app':
				return 'Mobile app';
```